sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/ThriftServerQueryTestSuite.scala

```diff
@@ -255,6 +255,22 @@ class ThriftServerQueryTestSuite extends SQLQueryTestSuite {
     }
   }
 
+  test("SPARK-29911: Uncache cached tables when session closed") {
+    val cacheManager = spark.sharedState.cacheManager
+    withJdbcStatement { statement =>
+      statement.execute("CACHE TABLE test1 AS SELECT 1")
+    }
+    // the cached data of local temporary view should be uncached
+    assert(cacheManager.isEmpty)
+    withJdbcStatement { statement =>
+      val globalTempDB = spark.sharedState.globalTempViewManager.database
+      statement.execute("CREATE GLOBAL TEMP VIEW test2 AS SELECT 1, 2")
+      statement.execute(s"CACHE TABLE $globalTempDB.test2")
+    }
+    // the cached data of global temporary view shouldn't be uncached
+    assert(!cacheManager.isEmpty)
```

Comment:
Shall we clean up `test2` in this test case?

Reply:
OK. But it can not be cleaned in `withGlobalTempView`. So I have to do it in a `try..finally` block.
```scala
    try {
      ...
    } finally {
      withJdbcStatement { statement =>
        statement.execute(s"UNCACHE TABLE IF EXISTS $globalTempDB.globalTempTbl")
      }
      assert(cacheManager.isEmpty)
    }
```

Reply:
Thanks!